interpreter/cling/lib/Interpreter/CIFactory.cpp

```diff
@@ -731,6 +774,50 @@ static void stringifyPreprocSetting(PreprocessorOptions& PPOpts,
     std::vector<const char*> argvCompile(argv, argv+1);
     argvCompile.reserve(argc+5);
 
+    // Variables for storing the memory of the C-string arguments.
+    // FIXME: We shouldn't use C-strings in the first place, but just use
+    // std::string for clang arguments.
+    std::string overlayArg;
+    std::string cacheArg;
+
+    // If user has enabled C++ modules we add some special module flags to the
+    // compiler invocation. C++ modules don't make any sense when we are anyway
+    // supposed to Lex the input, so let's not do it in the OnlyLex mode.
```

Comment:
Isn't `OnlyLex` the tab completion flag? In that case modules would help, right?

Reply:
Just to document what we discussed: It's  only used by the LinkdefReader from what I can see. And here we can't use modules (as the LinkdefReader can't parse headers because `__CINT__`is defined when we parse Linkdefs, and this define breaks user code).

But this check was indeed only back when we used the global ROOT_MODULES setting. so I think this check can indeed can go away so one can use this setting in cling.